src/ParseServer.js

```diff
@@ -83,45 +83,72 @@ class ParseServer {
     logging.setLogger(loggerController);
 
     // Note: Tests will start to fail if any validation happens after this is called.
-    databaseController
-      .performInitialization()
-      .then(() => hooksController.load())
-      .then(async () => {
+    (async () => {
+      try {
+        await databaseController.performInitialization();
+        await hooksController.load();
         if (schema) {
           await new DefinedSchemas(schema, this.config).execute();
         }
+        if (cloud) {
+          addParseCloud();
+          if (typeof cloud === 'function') {
+            await cloud(Parse);
+          } else if (typeof cloud === 'string') {
+            if (process.env.npm_package_type === 'module') {
+              await import(path.resolve(process.cwd(), cloud));
+            } else {
+              require(path.resolve(process.cwd(), cloud));
+            }
+          } else {
+            throw "argument 'cloud' must either be a string or a function";
+          }
+        }
         if (serverStartComplete) {
           serverStartComplete();
```

Comment:
use now "Promise.resolve(serverStartComplete())" to handle correctly a async serverStartComplete